### ISSUE TITLE
Consent not being recorded for any other user

### ISSUE THREAD

**Issue by antonskv** [docusign/eg-01-php-jwt#7]
Hi, 

Most of our integration is already done with admin user. But as soon as user switches to another user with another  DocuSign account, the consent becomes an infinite loop. User gets the link, authorizes, allows access to integrator application, then once anything else is done the consent pops up again. 

I tried it with your application of this repository just to make sure.  

This is the URL that is generated: 
https://account-d.docusign.com/oauth/auth?response_type=code&scope=signature&client_id=<INTEGRATOR_KEY_ID>&redirect_uri=<REDIRECT_URL>

I tried with both "signature" scope and "signature20%impersonation" scope in URL. Every time the consent required error is returned. 

This is the token that gets requested: 

Array
(
    [iss] => (INTEGRATOR_KEY)
    [sub] => (USER'S OWN API USERNAME)
    [aud] => account-d.docusign.com
    [scope] => signature impersonation
    [nbf] => 1582220855
    [exp] => 1582820855
)

Any idea why this keeps happening?

**Comment by LarryKlugerDS** [comment]
What does the redirect url redirect to? The application or a simple page that says “thank you for granting consent to the application.”

(Should be the latter.)

Consent is one time per person per integration key (client id).

Regards,
Larry

**Comment by antonskv** [comment]
Yeah, it's just a consent confirmation page saying "You're good to go". But when i run this repo code again, the consent pops up after authorization has been done. When we go to that URL, it does not show "<App Name> is requesting access" page anymore, it goes straight into redirect URL. Which makes me think that this specific user already authorized this specific integration key to work with their account. Yet, it keeps looping back consent_required error. Just to clarify, i'm using this repo code to make sure it's not an error in our own code.

**Comment by antonskv** [comment]
Oh and user's profile tab with connected apps, show this application already being authorized. If that helps any.

**Comment by antonskv** [comment]
Never mind, it was a typo inside the scope of original URL i pasted for consent. I guess API does not mind granting with invalid scopes requested. I do apologize for bothering you for nothing.

**Comment by LarryKlugerDS** [comment]
No worries! I’m glad it’s working!
